parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -258,16 +338,27 @@ public void writeTo(OutputStream out) throws IOException {
 
       value.mark();
       value.get(bytes).reset();
+      cachedBytes = bytes;
```

Comment:
The bytes can only be cached if the backing buffer isn't reused. If it is reused, then the backing buffer might change and cachedBytes could be stale.